lib/p2p/Pool.ts

```diff
@@ -172,7 +172,8 @@ class Pool extends EventEmitter {
       // check that this node is not ourselves, that it has listening addresses,
       // and that either we haven't heard of it, or we're not ignoring known nodes and it's not banned
       if (node.nodePubKey !== this.handshakeData.nodePubKey && node.addresses.length > 0 &&
-        (!this.nodes.has(node.nodePubKey) || (!ignoreKnown && !this.nodes.isBanned(node.nodePubKey)))) {
+        (!this.nodes.has(node.nodePubKey) || (!ignoreKnown && !this.nodes.isBanned(node.nodePubKey))) &&
+        !this.peers.has(node.nodePubKey)) {
         connectionPromises.push(this.tryConnectNode(node, retryConnecting));
```

Comment:
What do you try to achieve in this check?

Reply:
It's to ensure that we don't attempt a connection to a node that we're already connected to. I think it should've been in there before, based on the description of the method. And it should reduce logging messages from attempting to connect to peers we're already connected to.

Reply:
if this node connects with 3 other nodes and each of them inform him about a new node, does this check suppose to prevent him from trying to connect to that new nodes 3 times?

If yes - Not sure that it is working
if no - do we have an issue for that?

Reply:
Not exactly, I think #398 would achieve that. This check prevents you from attempting a connection to a peer you are *already* connected to, but not one that you are in the process of connecting to which can happen when you hear about the same peer multiple times almost simultaneously.